controllers/configurationpolicy_controller.go

```diff
@@ -387,13 +392,23 @@ func shouldEvaluatePolicy(policy *policyv1.ConfigurationPolicy, cleanupImmediate
 	} else if policy.Status.ComplianceState == policyv1.NonCompliant && policy.Spec != nil {
 		interval, err = policy.Spec.EvaluationInterval.GetNonCompliantInterval()
 	} else {
-		log.V(2).Info("The policy has an unknown compliance. Will evaluate it now.")
+		log.V(1).Info("The policy has an unknown compliance. Will evaluate it now.")
+
+		return true
+	}
+
+	usesSelector := policy.Spec.NamespaceSelector.MatchLabels != nil ||
+		policy.Spec.NamespaceSelector.MatchExpressions != nil ||
+		len(policy.Spec.NamespaceSelector.Include) != 0
+
+	if usesSelector && r.SelectorReconciler.HasUpdate(policy.Name) {
```

Comment:
I think this would be better since if the policy has no namespace selector it would just return false.
```suggestion
	if r.SelectorReconciler.HasUpdate(policy.Name) {
```

Reply:
My thought was to avoid `HasUpdate` when possible, because it requires getting a read lock.